File: TwoSum.java
import java.util.Map;
import java.util.HashMap;
public class TwoSum {
    public static void main(String[] args) {
        int[] test = new int[] {3, 2, 4};
        int target = 6;
        int[] result = twoSum(test, target);

        System.out.println(result[0] + " " + result[1]);
    }

    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (map.containsKey(complement)) {
                return new int[] {map.get(complement), i};
            }
            map.put(nums[i], i);
        }
        return new int[] {-1, -1};
    }
}

File: LeetcodeSolutions/Hashing/TwoSum.py
def twoSum(nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        map = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in map:
                return [map[complement], i]
            map[num] = i

nums = [2, 7, 11, 15]
target = 9
print(twoSum(nums, target))